pkg/exec/exec.go

```diff
@@ -21,7 +26,7 @@ func ExecuteCommand(client ExecClient, compInfo common.ComponentInfo, command []
 	reader, writer := io.Pipe()
 	var cmdOutput string
```

Comment:
The conventions for mutexes are that they are kept above the things they are guarding. Now you can create a struct called `SafeString` which holds the mutex and the string or as the mutex is not accessed externally just keep it above `cmdOutput`.